### ISSUE TITLE
Test2::Mock'ed objects don't pass an 'isa' test for the classes being mocked

### ISSUE THREAD

**Issue by shaunmlowry** [Test-More/Test2-Suite#124]
Consider the following classes:

```
package MyClass1;

use Moose;

has 'my_obj' => (
    is    => 'ro',
    required => 1,
    isa      => 'MyClass2',
);

1;
```

and

```
package MyClass2;

use Moose;

sub hello {
    print("hello!\n");
}

1;
```

The following code fails because the mocked object fails type validation:
```
use Test2::Mock;
use MyClass1;
use MyClass2;

my $obj_2 = Test2::Mock->new(
    class    => 'MyClass2',
    override => [
        hello => sub { print("goodbye\n"); },
    ],
);

my $obj_1 = MyClass1->new( my_obj => $obj_2 );
$obj_1->my_obj->hello();
```

**Comment by exodist** [comment]
The problem here is that Test2::Mock is not being used correctly. Specifically `Test2::Mock->new` returns a Test2::Mock instance, which is a mocking interface, it is not an instance of the class you are mocking. IE it is not supposed to work the way you use it here.

The correct way would be like this:

```
my $mock = Test2::Mock->new(...);
my $obj_2 = MyClass2->new;
my $obj_1 = MyClass->new(my_obj => $obj2);
$obj_1->my_obj->hello()
```

Test2::Mock takes control of the namespace you are mocking, and replaces the method in the symbol table. Once you define the mock you go ahead and use a real instance of the class you mocked. Once the $mock object goes out of scope the override is removed.

In other words, the object returned by Test2::Mock is an interface for overriding the symbol table. Once you have put your mocks in place you use a real instance of your original class.
